sdk/logical/event.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/vault/sdk/logical";
+
+package logical;
+
+// EventData contains event data in a CloudEvents container.
+message EventData {
+   	// EID identifies the event, cannot be an empty and is required. The
+	// combination of Source + EID must be unique. Events with the same Source +
+	// ID can be assumed to be duplicates by consumers.
+    // (Named EID to avoid conflict with ID method.)
+	string eid = 1;
+
+    // Arbitrary non-secret data. Optional.
+    bytes metadata = 2;
+
```

Comment:
I think most of the plugin-related fields below, I wouldn't want to have to trust/require the plugins to know and populate (e.g. especially mount path). It would be nice if the plugin events interface only exposed the fields we expect the plugin itself to fill in, and then populate the rest ourselves in Vault core (rather than overwriting fields the plugin may have already set).

Might it make sense to have an embedded proto struct within `EventData` to accommodate that? I haven't thought through all the implications so there may be reasons that would end up ugly.

Reply:
I think this is what Armon suggested, so I'll rework this to use a separate struct for plugin info.